core/math/SH.cpp

```diff
@@ -36,11 +36,13 @@ namespace MR
         "Each volume in the output image corresponds to a different spherical harmonic "
         "component. Each volume will correspond to the following: \n"
         "volume 0: l = 0, m = 0 ; \n"
-        "volume 1: l = 2, m = -2 (imaginary part of m=2 SH) ; \n"
```

Comment:
I think you may have misunderstood the intention of @jdtournier's initial wording here. The l and m information indicates the parameterisation, i.e. just like the volume number (as an alternative parameterisation) does as well.  The bit between brackets is merely a description of the definition for said parameters.  E.g., for l=2 and m=-2, the definition of the function implies that you have to compute the imaginary part of the l=2 and m=2 SH.  The confusion might stem from the different use of m as a parameter: the first m (in this example -2) is a parameter of the "modified" SH basis, i.e. the one we define here in; whereas the second m is a parameter of the "standard" or "unmodified" SH that is used "within" the definition of the "modified" SH basis.  My wording here is on purpose a bit informal, but I suppose you get the point.

Just as a more formal example, I think one of the first ones (or at least one of the most often referred to) is Maxime's early PhD paper where he used symmetrical SH to model ADCs: https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.20948 .  See equation 11 in there. It might not be exactly the basis used in mrtrix, but the general format of defining it is the same.  The original version of the documentation here defines the basis in function of volume number, similar to how Maxime's paper defines it in function of parameter "j".  The "j" parameter (or volume index) itself is defined in function of l and m (see the sentence before equation 11 in the paper). Finally there is equation 11 itself, which explains how this is then actually evaluated, and part of that is an "unmodified" SH evaluation; but for some values of m you have to take the real part, and for others the imaginary part; and this choice indeed depends on the sign of m.

So long story short: the original documentation here is very much in line to the way in which this is usually formally defined in the literature.  But the brevity of the documentation might lead to confusion.  I would therefore personally argue the brevity is the problem; but rather than making individual command documentation excessively long, it might be wise to keep it short on purpose, and include a reference to another place (this might e.g. be a documentation page), where there is enough space / flexibility / words to describe this in detail in a way that doesn't allow for confusion.  My final 2 cents would be that most users don't need this in detail in the command description; it's a lot of lines that they probably want to scroll past each time again just to look up the name of an option of said command, in my experience the most common use of the inline docs for many frequent users.  Users that require this description, are probably at least already trying to port information between software packages, or are implementing stuff on their own.  They would probably look into more detailed docs anyway, rather than going by a command description.